base/math.hpp

```diff
@@ -102,8 +102,8 @@ template <typename T> inline T sq(T const & x)
   return x * x;
 }
 
-template <typename T, typename TMin, typename TMax>
-inline T clamp(T x, TMin xmin, TMax xmax)
+template <typename T>
+inline T clamp(T const x, T const xmin, T const xmax)
```

Comment:
const is redundant here. in general it is ok to have const here but then reasonable question arises - why const is not used in other functions - cyclicClamp, between_ and etc. would be good to make it in the one style.